crates/swc_ecma_minifier/src/compress/optimize/sequences.rs

```diff
@@ -1068,11 +1068,15 @@ where
 
         match e {
             Expr::Ident(e) => {
-                if e.span.ctxt == self.expr_ctx.unresolved_ctxt
-                    && !is_global_var_with_pure_property_access(&e.sym)
-                {
-                    log_abort!("Undeclared");
-                    return false;
+                if e.span.ctxt == self.expr_ctx.unresolved_ctxt {
+                    return if self.options.pristine_globals
+                        && is_global_var_with_pure_property_access(&e.sym)
+                    {
+                        true
+                    } else {
+                        log_abort!("Undeclared");
+                        return false;
+                    };
```

Comment:
Although it's tempting to push this skip to all non-mutated idents, but considering following case
```js
var a = b.x
b.inc()
console.log(a)
```